internal/pkg/config/config_test.go

```diff
@@ -40,15 +41,15 @@ func runTest(pass *analysis.Pass) (interface{}, error) {
 	}
 
 	for _, f := range in.SrcFuncs {
-		if conf.IsSinkFunction(f) {
+		if conf.IsSink(utils.DecomposeFunction(f)) {
```

Comment:
I think this inlined call looks quite nice.

Reply:
On the other hand, I'm not sure I like the naming. A few thoughts:
1. I think the name `utils` has overstayed its welcome, and it would help to have a more specific name. (I think it might make sense to have two packages, `ssautils` and `typeutils`, but that wouldn't be a huge gain).
2. The name `DecomposeFunction` sounds quite nice to me, but I think it is too general to accurately describe the behavior. Really, this is just grabbing the name, recv and pkg of the function, but to me `Decompose` hints that something deeper is going on. This comment also applies to `DecomposeType`. (A suggestion: maybe it would read better to have a function named `DescribeFunction` that would produce a `functionDescriptor` struct with the relevant fields, and have the config take that instead.)

Reply:
I agree that `Decompose*` is not a great name, and that `utils` is also a bit too ambiguous.  Will update soon.

I agree with making a more generic interface to the `type Config` itself, but that feels like a future PR.

Reply:
I've been staring at this for a short while now, and for the life of me, I can't come up with a better name than `Decompose`.  `FunctionDetails` seems equally generic.  `PathTypeAndNameStrings(f)` seems cumbersome.  `IDStrings`?  I don't know.

As for `utils`, I'm just going to end up pulling three functions into their own sub-utils.  Since they're intended to massage the `config` interface, I suppose they could live in `config`.  That's where there were during iteration, but it felt like violating the "`config` shouldn't know about `ssa`" ideal.

Do you have any suggestions for names?

Reply:
I've thought about the naming some more and sadly I don't have a very solid suggestion to make. I think `Decompose*` is fine. Having a `functionDescriptor` struct could be a nice approach, but maybe not.